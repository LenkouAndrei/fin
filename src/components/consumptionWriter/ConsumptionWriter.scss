.w-400 {
    width: 400px;
}

.h-center {
    margin: 0 auto;
}

.list-item {
    margin: 5px 0;
}