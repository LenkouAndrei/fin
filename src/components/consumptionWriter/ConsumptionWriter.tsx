import { Button } from "../button/Button";
import { Input } from "../input/Input";
import "./ConsumptionWriter.scss";

export const ConsumptionWriter: React.FC = () => {
  return (
    <ul className="list w-400 h-center">
      <li className="list-item">
        <Input>How much</Input>
      </li>
      <li className="list-item">
        <Input>For what</Input>
      </li>
      <li className="list-item">
        <Button>Submit</Button>
      </li>
    </ul>
  );
};
