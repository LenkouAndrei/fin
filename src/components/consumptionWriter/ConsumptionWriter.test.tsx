import { render, screen } from '@testing-library/react';
import { ConsumptionWriter } from './ConsumptionWriter';

describe('<ConsumptionWriter />', () => {
  it('should render ConsumptionWriter', () => {
    const mockCallback = jest.fn();
    const { asFragment } = render(<ConsumptionWriter updateData={mockCallback}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should use callback when submit clicked', () => {
    const mockCallback = jest.fn();
    render(<ConsumptionWriter updateData={mockCallback}/>);
    const submitBtn = screen.getByText('Submit');
    submitBtn.click();
    expect(mockCallback).toHaveBeenCalled();
  });
});
