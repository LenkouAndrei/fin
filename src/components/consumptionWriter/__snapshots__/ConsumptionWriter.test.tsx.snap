// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsumptionWriter /> should render ConsumptionWriter 1`] = `
<DocumentFragment>
  <ul
    class="list w-400 h-center"
  >
    <li
      class="list-item"
    >
      <input
        class="ipt"
        placeholder="How much"
        type="text"
        value=""
      />
    </li>
    <li
      class="list-item"
    >
      <input
        class="ipt"
        placeholder="What for"
        type="text"
        value=""
      />
    </li>
    <li
      class="list-item"
    >
      <button
        class="btn regular   medium"
      >
        Submit
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
